mbuild/lattice.py

```diff
@@ -110,42 +108,51 @@ class Lattice(object):
     input.
 
     Generating FCC Copper cell with lattice_vectors instead of angles
+
     >>> import mbuild as mb
     >>> copper = mb.Compound(name='Cu')
-    >>> lattice_vector = ( [1, 0, 0], [0, 1, 0], [0, 0, 1])
-    >>> spacings = [.36149, .36149, .36149]
+    >>> lattice_vector = [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
+    >>> spacing = [.36149, .36149, .36149]
     >>> copper_locations = [[0., 0., 0.], [.5, .5, 0.],
     ...                     [.5, 0., .5], [0., .5, .5]]
     >>> basis = {'Cu' : copper_locations}
-    >>> copper_lattice = mb.Lattice(spacings, dimension=3,
-    ...                           lattice_vectors=lattice_vector,
-    ...                           basis_atoms=basis)
+    >>> copper_lattice = mb.Lattice(lattice_spacing = spacing,
+    ...                             lattice_vectors=lattice_vector,
+    ...                             lattice_points=basis)
     >>> copper_dict = {'Cu' : copper}
-    >>> copper_cell = copper_lattice.populate(x=3, y=3, z=20,
+    >>> copper_pillar = copper_lattice.populate(x=3, y=3, z=20,
     ...                                       compound_dict=copper_dict)
 
-    TODO(Justin Gilmer) : Print function to display info about Lattice (repr)
-    TODO(Justin Gilmer) : inheritance(Cubic, orthorhombic, hexangonal)
-    TODO(Justin Gilmer) : orientation functionality
+    Generating the 2d Structure Graphene carbon backbone
+
+    >>> import mbuild as mb
+    >>> carbon = mb.Compound(name='C')
+    >>> angles = [90, 90, 120]
+    >>> carbon_locations = [[0, 0, 0], [2/3, 1/3, 0]]
+    >>> basis = {'C' : carbon_locations}
+    >>> graphene = mb.Lattice(lattice_spacing=[.2456, .2456, 0],
+    ...                        angles=angles, lattice_points=basis)
+    >>> carbon_dict = {'C' : carbon}
+    >>> graphene_cell = graphene.populate(compound_dict=carbon_dict,
+    ...                                   x=3, y=3, z=1)
+
     """
 
-    def __init__(self, lattice_spacings, dimension=None,
-                 lattice_vectors=None, basis_atoms=None,
-                 angles=None):
+    def __init__(self, lattice_spacing=None, lattice_vectors=None,
+                 lattice_points=None, angles=None):
         super(Lattice, self).__init__()
-        self.lattice_spacings = None
-        self.dimension = None
+        self.dimension = 3
```

Comment:
Does it make sense to keep `dimension` as an attribute if it will always be 3?

Reply:
I kept it in since I use the dimension as a shape constraint many times in the validation steps. However, if for some reason the dimensionality of the lattice needed to change, the underlying code should not need much rework since I refer to it from its attribute.